### PR TITLE
Fix the core-auth dependency

### PR DESCRIPTION
**Pull request by sarangan12**
This is to fix the issue reported in https://github.com/Azure/autorest.typescript/issues/1066.

Minor fix in the logic to add core-auth in the package json file. 

@joheredi @deyaaeldeen @dw511214992 Please review and approve.

@ramya-rao-a FYI.....

### CODE REVIEW

**Review by dw511214992 — APPROVED** [review]

File: src/generators/static/packageFileGenerator.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ function restLevelPackage(packageDetails: PackageDetails) {
     },
     dependencies: {
       "@azure-rest/core-client": "1.0.0-beta.4",
-      "@azure/core-auth": "^1.1.4",
+      "@azure/core-auth": "^1.3.0",
       "@azure/core-rest-pipeline": "^1.0.4"
```
Comment (maorleger) [inline src/generators/static/packageFileGenerator.ts:92]: `@azure/core-rest-pipeline` should be updated to "^1.1.0" as well - can we do it as part of this PR or a separate PR?

Comment (sarangan12) [reply src/generators/static/packageFileGenerator.ts:92]: I can combine it with this PR. Will update it shortly

File: test/integration/generated/iotspaces/package.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  "name": "iotspaces",
+  "sdk-type": "management",
+  "author": "Microsoft Corporation",
+  "description": "A generated SDK for IoTSpacesClient.",
+  "version": "1.0.0-preview1",
+  "engines": { "node": ">=12.0.0" },
+  "dependencies": {
+    "@azure/core-client": "^1.1.2",
+    "@azure/core-auth": "^1.3.0",
+    "@azure/core-rest-pipeline": "1.0.0-beta.2",
```
Comment (deyaaeldeen) [inline test/integration/generated/iotspaces/package.json:11]: why are generating a beta version here?

File: test/integration/generated/iotspaces/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "module": "es6",
+    "moduleResolution": "node",
+    "strict": true,
+    "target": "es6",
```
Comment (deyaaeldeen) [inline test/integration/generated/iotspaces/tsconfig.json:6]: No need to be fixed in this PR but I think we can target ES2017 everywhere now to be consistent with Track 2 packages in the mono repo.

File: test/utils/test-swagger-gen.ts [hunk 4]
```diff
@@ -702,6 +702,15 @@ const testSwaggers: { [name: string]: SwaggerConfig } = {
     allowInsecureConnection: true,
     addCredentials: false
   },
+  iotspaces: {
+    swaggerOrConfig: "test/integration/swaggers/iotspaces.json",
```
Comment (deyaaeldeen) [inline test/utils/test-swagger-gen.ts:706]: what is interesting about this swagger to be added in our tests?

File: src/generators/static/packageFileGenerator.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/generators/static/packageFileGenerator.ts:92]: 1.1.0 of core-rest-pipeline is not published yet...

Comment (sarangan12) [reply src/generators/static/packageFileGenerator.ts:92]: Spoke offline with Maor. Will keep core-rest-pipeline to 1.0.4 all through for this week.

File: test/integration/generated/iotspaces/package.json (hunk 2 above)
Comment (sarangan12) [reply test/integration/generated/iotspaces/package.json:11]: I think this is a left over. Changed it now

File: test/integration/generated/iotspaces/tsconfig.json (hunk 3 above)
Comment (sarangan12) [reply test/integration/generated/iotspaces/tsconfig.json:6]: Will do it in a seperate PR

**Pull request merged**